hypershift-operator/controllers/hostedcluster/internal/platform/platform.go

```diff
@@ -55,6 +56,9 @@ type Platform interface {
 	// DeleteCredentials is responsible for deleting resources related to platform credentials
 	// So they won't leak on upon hostedcluster deletion
 	DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error
+
+	// ReconcileExternalCCM is responsible for installing cloud controller manager with respect to hosted cluster's platform
+	ReconcileExternalCCM(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, createOrUpdate upsert.CreateOrUpdateFN, kubeConfig *corev1.Secret, ccmImage string) error
```

Comment:
I am not a huge fan of extending this interface for every single thing some provider might want, as it then forces all provides to implement that. As long as there is no strong ordering requirement, how about simply making it a generic `ReconcileProviderManagementClusterResources`?

Reply:
Moved the CCM deployment as part of CPO controller. Removed this.